aten/src/ATen/native/metal/MetalPrepackOpRegister.cpp

```diff
@@ -37,7 +37,7 @@ c10::intrusive_ptr<Conv2dOpContext> unpack(
       output_max);
 }
 
-TORCH_LIBRARY(metal, m) {
+TORCH_LIBRARY_2D(metal, m) {
```

Comment:
I believe TORCH_LIBRARY is a predefined macro? can we add a 2D suffix here?

Reply:
cc @xta0 can you review the changes related to metal?

Reply:
This thread is a dupplicate of this one https://github.com/pytorch/pytorch/pull/50003#discussion_r572928088. Let's keep talking here.

So yeah, I'm not at all sure I can do that.

Reply:
`TORCH_LIBRARY_2D ` seems not right. If you want to add conv3d override for metal, you would do something like 

```
TORCH_LIBRARY_IMPL(aten, Metal, m) {
  m.impl("conv3d", TORCH_FN(conv3d));
}
```

A bit context here, Metal is currently being used only on mobile, and the GPU ops are registered as custom ops that are not part of the PyTorch core. 
The changes here will definitely break things in production(FB, Instagram, Messenger, etc) within Facebook.

Reply:
Thank you. So I'll simply keep the usual naming and add the _3D version.

Reply:
Thank you. I'm gunning for more enthusiasm for these types of layers so i'm inclined to do it your way.